src/qt/intro.cpp

```diff
@@ -130,30 +130,14 @@ Intro::Intro(QWidget *parent, uint64_t blockchain_size, uint64_t chain_state_siz
     );
     ui->lblExplanation2->setText(ui->lblExplanation2->text().arg(PACKAGE_NAME));
 
-    uint64_t pruneTarget = std::max<int64_t>(0, gArgs.GetArg("-prune", 0));
+    pruneTarget = std::max<int64_t>(0, gArgs.GetArg("-prune", 0));
     if (pruneTarget > 1) { // -prune=1 means enabled, above that it's a size in MB
         ui->prune->setChecked(true);
         ui->prune->setEnabled(false);
     }
-    ui->prune->setText(tr("Discard blocks after verification, except most recent %1 GB (prune)").arg(pruneTarget ? pruneTarget / 1000 : 2));
-    requiredSpace = m_blockchain_size;
-    QString storageRequiresMsg = tr("At least %1 GB of data will be stored in this directory, and it will grow over time.");
-    if (pruneTarget) {
-        uint64_t prunedGBs = std::ceil(pruneTarget * 1024 * 1024.0 / GB_BYTES);
-        if (prunedGBs <= requiredSpace) {
-            requiredSpace = prunedGBs;
-            storageRequiresMsg = tr("Approximately %1 GB of data will be stored in this directory.");
-        }
-        ui->lblExplanation3->setVisible(true);
-    } else {
-        ui->lblExplanation3->setVisible(false);
-    }
-    requiredSpace += m_chain_state_size;
-    ui->sizeWarningLabel->setText(
-        tr("%1 will download and store a copy of the Bitcoin block chain.").arg(PACKAGE_NAME) + " " +
-        storageRequiresMsg.arg(requiredSpace) + " " +
-        tr("The wallet will also be stored in this directory.")
-    );
+    setSizeWarningLabel();
+    pruneTarget = pruneTarget ? pruneTarget / 1000 : 2;
```

Comment:
`pruneTarget` variable holds semantically different values at different points: (1) pruned target in mebibytes (MiB), and (2) pruned target in gigabytes (10^6 bytes). IMO, such semantic overloading should be avoided.

Please note, that the binary coefficient is used in master branch: https://github.com/bitcoin/bitcoin/blob/46d6930f8c7ba7cbcd7d86dd5d0117642fcbc819/src/qt/intro.cpp#L142

`-prune` option also uses a binary prefix for its value -- MiB.

Reply:
ACK

Reply:
IIRC it still uses megabyte internally but only gigabyte for the display in the GUI because otherwise it could get strange looking

Reply:
ACK on the overloading problem.
I will multiply with 1000 instead

Reply:
Eh? Why is this comment marked as resolved?